### ISSUE TITLE
[makeotfexe] Improve overflow error messages to help diagnose issues

### ISSUE THREAD

**Issue by davelab6** [adobe-type-tools/afdko#53]
Following #52, I suggest that table overflow error messages should warn users to check for 'upstream' overflows.

**Comment by miguelsousa** [comment]
@tntypography can clarify, but my understanding is that there was no overflow happening in the kerning alone. The overflow only occurred when more lookups (the `mark` and `mkmk`) where added to the GPOS table.

**Comment by tntypography** [comment]
@miguelsousa that is correct. I suppose my grasp for what actually happens in an overflow is not sufficient to understand what was going on with my fonts. All I know is that reducing the size of the kern feature prevented the overflow happening in mark and mkmk features.

**Comment by davelab6** [comment]
> reducing the size of the kern feature prevented the overflow happening in mark and mkmk features.

Right; I'm suggesting that when an overflow happens in mark and mkmk features, the error message should mention that reducing the size of the kern feature can resolve this.

**Comment by davelab6** [comment]
> reducing the size of the kern feature prevented the overflow happening in mark and mkmk features.

Right; I'm suggesting that when an overflow happens in mark and mkmk features, the error message should mention that overflows can be caused 'upstream,' so reducing the size of other features (eg the kern feature) can resolve overflows.

**Comment by miguelsousa** [comment]
Got it. I'd rephrase the suggestion to *when an overflow happens, look upstream to see if there are any lookups of another feature; if so, suggest trying to reduce those*.

**Comment by tntypography** [comment]
Either of those would have helped me. is there any documentation of when an overflow can happen? I am not aware of anything like that.

**Comment by miguelsousa** [comment]
@tntypography overflows are implicit in the OpenType spec. Whenever the tool has e.g. 36K of data to put in a field that is limited to maximum of 32K, an overflow message happens and the font is not built (because if it were built it would be violating the spec).

**Comment by davelab6** [comment]
That phrasing is good! :)

**Comment by tntypography** [comment]
Thanks Miguel, that helps a little. Now I'll just need to learn to translate kerning pairs & classes into kb... ;)

**Comment by miguelsousa** [comment]
In #631 improvements were made to all overflow-related messaging coming from `makeotfexe`. The messages still do not provide any guidance on how to resolve the problems, but that's as good as we can make them for now.
